packages/liveblocks-node/src/client.ts

```diff
@@ -194,6 +294,379 @@ export class Liveblocks {
     }
   }
 
+  /* -------------------------------------------------------------------------------------------------
+   * Room
+   * -----------------------------------------------------------------------------------------------*/
+
+  /**
+   * Returns a list of your rooms. The rooms are returned sorted by creation date, from newest to oldest. You can filter rooms by metadata, users accesses and groups accesses.
+   * @param params.limit (optional) A limit on the number of rooms to be returned. The limit can range between 1 and 100, and defaults to 20.
+   * @param params.startingAfter (optional) A cursor used for pagination. You get the value from the response of the previous page.
+   * @param params.userId (optional) A filter on users accesses.
+   * @param params.metadata (optional) A filter on metadata. Multiple metadata keys can be used to filter rooms.
+   * @param params.groupIds (optional) A filter on groups accesses. Multiple groups can be used.
+   * @returns A list of rooms.
+   */
+  public async getRooms(
+    params: {
+      limit?: number;
+      startingAfter?: number;
+      metadata?: RoomMetadata;
+      userId?: string;
+      groupIds?: string;
+    } = {}
+  ): Promise<{
+    nextPage: string | null;
+    data: RoomInfo[];
+  }> {
+    const path = url`v2/rooms`;
+
+    const queryParams = new URLSearchParams();
+    if (params.limit) {
+      queryParams.set("limit", params.limit.toString());
+    }
+    if (params.startingAfter) {
+      queryParams.set("startingAfter", params.startingAfter.toString());
+    }
+    if (params.userId) {
+      queryParams.set("userId", params.userId);
+    }
+    if (params.groupIds) {
+      queryParams.set("groupIds", params.groupIds);
+    }
+    if (params.metadata) {
+      Object.entries(params.metadata).forEach(([key, val]) => {
+        const value = Array.isArray(val) ? val.join(",") : val;
+        queryParams.set(`metadata.${key}`, value);
+      });
+    }
```

Comment:
With the introduction of the new helper, you should be able to write this more succinctly (but as safely), using the following:

```suggestion
    const queryParams = {
      limit: params.limit,
      startingAfter: params.startingAfter,
      userId: params.userId,
      groupIds: params.groupIds,

      // "Flatten" {metadata: {foo: "bar"}} to {"metadata.foo": "bar"}
      ...Object.fromEntries(
        Object.entries(params.metadata ?? {}).map(([key, val]) => {
          const value = Array.isArray(val) ? val.join(",") : val;
          return [`metadata.${key}`, value];
        })
      ),
    };
```

Reply:
This one I did to demonstrate how we can also use `URLSearchParams` with the new `GET` method. There is another place I use the `Record<string, string | number | null | undefined>` type instead but makes sense to keep it consistent in all places.